sdks/go/pkg/beam/core/runtime/xlangx/expansionx/process.go

```diff
@@ -30,20 +31,39 @@ type ExpansionServiceRunner struct {
 	serviceCommand *exec.Cmd
 }
 
+func findOpenPort() (int, error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return 0, err
+	}
+	defer listener.Close()
+	return listener.Addr().(*net.TCPAddr).Port, nil
+}
+
 // NewExpansionServiceRunner builds an ExpansionServiceRunner struct for a given gradle target and
 // Beam version and returns a pointer to it.
-func NewExpansionServiceRunner(jarPath, servicePort string) *ExpansionServiceRunner {
+func NewExpansionServiceRunner(jarPath, servicePort string) (*ExpansionServiceRunner, error) {
 	if servicePort == "" {
-		servicePort = "8097"
+		port, err := findOpenPort()
+		if err != nil {
+			return nil, fmt.Errorf("failed to find open port for service, got %v", err)
+		}
+		servicePort = fmt.Sprintf("%d", port)
 	}
 	serviceCommand := exec.Command("java", "-jar", jarPath, servicePort)
-	return &ExpansionServiceRunner{jarPath: jarPath, servicePort: servicePort, serviceCommand: serviceCommand}
+	return &ExpansionServiceRunner{jarPath: jarPath, servicePort: servicePort, serviceCommand: serviceCommand}, nil
 }
 
 func (e *ExpansionServiceRunner) String() string {
 	return fmt.Sprintf("JAR: %v, Port: %v, Process: %v", e.jarPath, e.servicePort, e.serviceCommand.Process)
 }
 
+// GetPort returns the formatted port the ExpansionServiceRunner is set to start the expansion
+// service on.
+func (e *ExpansionServiceRunner) GetPort() string {
```

Comment:
s/GetPort/Endpoint

What's being returned isn't just the port, it's the address + port pair. If an ipaddress were used, it would be called a "SocketAddress" instead. Since it's not an IP address, the best term is Endpoint.

Finally, idiomatic Go drops the "Get" prefix unless the name would conflict with an exported field or method.

Reply:
Interesting point regarding getters in Go. Changed. 

Although wouldn't an exported field be directly accessible anyway? My intuition here is that unexported fields have a lot of use as private fields in a getter/setter structure and that an explicit Get/Set username isn't a bad thing. This is more a comment on general Go style, not this particular instance